### PR TITLE
(1) fix: clean up modal logic surrounding admin assessments

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link

[Add loading state + toasts to all modals](https://www.notion.so/uwblueprintexecs/Add-loading-state-toasts-to-all-modals-7d8198e55ee545ebbe1f096740937b67)

## Implementation description
**Strongly** suggest [reviewing by commit](https://github.com/uwblueprint/jump-math/pull/569/commits/48a910460943083eba1f2a17fc4dfdbee47f2f81?w=1)! The diff will be way smaller that way.

- Update generic `Modal` component to have generified error handling
- Update `ArchiveAssessmentModal` to use new props
- Update `DeleteAssessmentModal` to use new props
- Update `DuplicateAssessmentModal` to use new props
- Update `PublishAssessmentModal` to use new props
- Update `UnarchiveAssessmentModal` to use new props
- Fix name of `PopoverButton` component
- Fix and reduce modal/submission logic in assessment editor

## Steps to test
View the relevant pages and test various modal actions (e.g., cause an error by turning off your "internet" via the network tab).

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/assessment-creation/AssessmentEditorHeader.tsx [hunk 1]
```diff
@@ -142,7 +142,7 @@ const AssessmentEditorHeader = ({
                     name="Archive"
                     onClick={() => {
                       onClosePopover();
-                      handleArchive();
+                      onArchiveModalOpen();
```
Comment (jfdoming) [inline frontend/src/components/admin/assessment-creation/AssessmentEditorHeader.tsx:145]: I don't think we need to be validating on archive, which is why I removed the validation check here!

**Review by carissa-tang — COMMENTED** [review]
is this the intended behaviour? to have both toasts
image

**Review by carissa-tang — APPROVED** [review]

File: frontend/src/components/common/popover/PopoverButton.tsx [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
   name: string;
-  onClick: React.MouseEventHandler<HTMLButtonElement>;
-}
-const EditStatusButton = ({
-  name,
-  "aria-label": ariaLabel,
-  onClick,
-}: EditStatusButtonProps): React.ReactElement => {
+};
+const PopverButton = ({ name, ...props }: PopverButtonProps): ReactElement => {
```
Comment (carissa-tang) [inline frontend/src/components/common/popover/PopoverButton.tsx]:
```suggestion
const PopoverButton = ({ name, ...props }: PopverButtonProps): ReactElement => {
```

**Comment by jfdoming** [comment]
Note: I will not be merging this yet since there will be a bunch of "Operation Successful" toasts everywhere once I do, and we are supposed to have a design review today! However, I'll consider this "done" and start opening additional follow-up PRs. I'll merge this after the review.

**Pull request merged**